### DISCUSSION TITLE
Refused to connect to 'http://npm.xxx.com/-/verdaccio/data/packages' because it violates the following Content Security Policy directive: "connect-src 'self'"

### DISCUSSION THREAD

**Discussion by nicky132** [verdaccio/verdaccio#4009] [Q&A]
**Your Environment**
 * **verdaccio version**: verdaccio/verdaccio   latest
 * **node version** [v16.17.0]:
 * **package manager**: [npm@8.15.0]
 * **os**: [windows@10]
 * **platform**: [docker]

**Describe the bug**

if i visit verdaccio via ip address,then i got error blow.but npm.xxx.com can be also used as well ,nothing wrong with it.
i hope i can use both  npm.xxx.com and ip to visit my npm mirror as well ! 
ERROR:
![1693825385126_DDA405DE-8ECA-4b3a-AC96-A288CB9D9AA4](https://github.com/verdaccio/verdaccio/assets/34881461/555cdcb3-52df-4e90-8c6c-97c77ea22824)

![1693824115646_AE9066E1-6D6C-4c17-9A30-A8F618A810FF](https://github.com/verdaccio/verdaccio/assets/34881461/1a01cc84-87ba-423f-b309-695a090c8b48)
GOOD:
![image](https://github.com/verdaccio/verdaccio/assets/34881461/b43088a1-6f85-4851-af1c-8d90a3e35ee6)

CONFIG:
mine images:
![1693824463047_9088EAF6-1335-477b-9CFB-14A703715D11](https://github.com/verdaccio/verdaccio/assets/34881461/ba24e521-672f-4865-a4c3-89adea440cd1)

mine config.yaml:
![1693825956784_11EAC175-520A-45b3-8BBD-A3FA5338943B](https://github.com/verdaccio/verdaccio/assets/34881461/e17640b3-0f6d-4567-814a-8eb9f2b590f4)

mine nginx:
![1693825252269_5291371C-8E42-4d3d-BCB2-BBD1F0416674](https://github.com/verdaccio/verdaccio/assets/34881461/8ac42009-cda1-4cb2-b487-d894339ef777)

**To Reproduce**

**Expected behavior**

Both works well,
if i visit verdaccio via ip address,then i got error blow.but npm.xxx.com can be also used as well ,nothing wrong with it.
i hope i can use both  npm.xxx.com and ip to visit my npm mirror as well ! 

**Environment information**

**Debugging output**
Refused to connect to 'http://npm.xxx.com/-/verdaccio/data/packages' because it violates the following Content Security Policy directive: "connect-src 'self'".

(anonymous) @ main.a8b34a3843b7ceb4e197.js:2
main.a8b34a3843b7ceb4e197.js:2 Refused to connect to 'http://npm.xxx.com/-/verdaccio/data/packages' because it violates the document's Content Security Policy.

**Contribute to Verdaccio**

**Comment by juanpicado** [comment]
Right now, so far you can use one option, either IP or domain, but no both, this has been asked already before, just to give you a quick answer.

**Reply by nicky132** [reply]
Thanks,by the way. if i want to use both,how can i do it.i really need it both

**Reply by nicky132** [reply]
and pls give me another answer you mentioned before,i pleased to check it out,and i really want to solve the problem

**Reply by leookun** [reply]
proxy_set_header rewrite host to  your domain ,should solve your problem

**Reply by nicky132** [reply]
As you can see, i have already rewrite "proxy_set_header       Host $http_host;" on my nginx.conf file.but nothing changed.
Am i right?

**Reply by xiuxiumomo** [reply]
能解决吗？服务器不能开端口号，只能域名访问，会有这个bug。

**Reply by xiuxiumomo** [reply]
> As you can see, i have already rewrite "proxy_set_header Host $http_host;" on my nginx.conf file.but nothing changed. Am i right?

我找打解决方法了，设置一下config/config.yaml  设置 html_cache false 重启一下服务就不会出现了
